source/howto/new-user-guide/yaml.rst

```diff
@@ -0,0 +1,114 @@
+##########################################
+
+Valohai is completely technology agnostic. You can develop in notebooks or scripts in a language and framework of your choice.
+
+.. admonition:: valohai.yaml
+  :class: important
+
+  Valohai expects that you have a :ref:`yaml` in the root of your repository. The file defines what kind of jobs can be executed inside your project, and the different properties of each job type (these are called steps in Valohai). 
+
+  You can either write the ``valohai.yaml`` by hand or if you're using Python you can use the :ref:`valohai-utils` define the configuration file in your code.
+
+
+Let's start by bringing just a single job type to Valohai.
+
+  
+.. tab:: valohai-utils (Python)
+
+  Open your existing Python file and define a Valohai step inside it:
+
+  .. code-block:: python
+
+    import python
+
+    # Prepares a new step called train in valohai.yaml
+    # Every time you run train in Valohai this Python script will be executed
+    valohai.prepare(step="train")
+
+  You can now generate a ``valohai.yaml`` configuration file that includes the ``train`` step by running:
+
+  .. code-block:: bash
+
+    vh yaml step myfilename.py
+
+  This will create:
+
+  * a ``valohai.yaml`` file with the ``train`` step that will run:
+
+    * ``pip install -r requirements.txt``
+    *  your Python script.
+
+  * a ``requirements.txt`` with ``valohai-utils``
+    
+    * If you have an existing ``requirements.txt`` file the ``valohai-utils`` line will be just added there.
+
+  By default, your step will use a standard Python Docker image and install everything from ``requirements.txt`` during runtime.
+
+  You could also specify a different Docker image by passing the image name to ``prepare``.
+
+  .. code-block:: python
+
+    import python
+
+    # Prepares a new step called train in valohai.yaml
+    # Every time you run train in Valohai this Python script will be executed
+    # inside a GPU enabled Tensorflow 2.6.1 image
+    valohai.prepare(step="train", image="tensorflow/tensorflow:2.6.1-gpu")
+
+  and then rerun ``vh yaml step`` to update your ``valohai.yaml``.
+
+  .. code-block:: bash
+
+    vh yaml step myfilename.py
+
+  You can now upload your local code to Valohai and run the ``train`` step and define a Docker image you'd like to use. 
+
+.. tab:: Other
+
+    Create a new file called ``valohai.yaml`` and define a :ref:`step` inside it.
+
+    In our example below we're defining two commands to be ran every time I run the step called ``train``. You can remove the pip install command, if you don't have a need for it.
+
+    .. code-block:: yaml
+
+        - step:
+            name: train
+            image: tensorflow/tensorflow:2.6.1-gpu
+            command:
+              - pip install -r requirements.txt
+              - python myfilename.py
+
+    ..
+
+    Create a ``valohai.yaml`` configuration file and define your step in it:
+
+
+.. code-block:: bash
+
+  vh execution run --adhoc train
```

Comment:
I'll keep this but move the 
"You can now upload your local code to Valohai and run the ``train`` step inside the Docker container you just defined." to be infront of it, instead of just in the first tab